lib/macho/load_commands.rb

```diff
@@ -268,7 +268,8 @@ def initialize(lc, lc_str)
         if view
           lc_str_abs = view.offset + lc_str
           lc_end = view.offset + lc.cmdsize - 1
-          @string = view.raw_data.slice(lc_str_abs..lc_end).delete("\x00")
+          raw_string = view.raw_data.slice(lc_str_abs..lc_end)
+          @string = raw_string.split("\x00").first
```

Comment:
I think `MalformedLCStrError` or `LCStrMalformedError` better captures the nature of the error.